charts/mattermost-operator/templates/mattermost-operator/deployment.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if .Values.mattermostOperator.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "mattermost-operator.name" . }}
```

Comment:
missing labels 
and for all the other manifests I recommend to follow this guide for the labels and annotations
https://helm.sh/docs/chart_best_practices/labels/

Reply:
Thanks added labels :)